cpu/sam0_common/periph/spi.c

```diff
@@ -126,8 +126,11 @@ int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
     /* power on the device */
     poweron(bus);
 
-    /* configuration did not change */
+    /* configuration did not change but device is powered down */
     if (dev(bus)->BAUD.reg == baud && dev(bus)->CTRLA.reg == ctrla) {
+        /* enable the device */
+        dev(bus)->CTRLA.reg |= SERCOM_SPI_CTRLA_ENABLE;
+
```

Comment:
Please also add
`while (dev(bus)->SYNCBUSY.reg & SERCOM_SPI_SYNCBUSY_ENABLE) {}`

Reply:
Or to avoid duplicating those two lines, change the if condition to

```C
if (dev(bus)->BAUD.reg != baud || dev(bus)->CTRLA.reg != ctrla) {
    /* disable the device */
    dev(bus)->CTRLA.reg &= ~(SERCOM_SPI_CTRLA_ENABLE); // is this even still needed anymore?
    while (dev(bus)->SYNCBUSY.reg & SERCOM_SPI_SYNCBUSY_ENABLE) {}

    dev(bus)->BAUD.reg = baud;
    dev(bus)->CTRLA.reg = ctrla;
}

Reply:
@benpicco CTRLA is enable-protected and your matching condition does not assert that the device is disabled, so we should disable it to be sure.

Reply:
I agree with the suggested changes and just pushed an update using Ben's rewrite, including always disabling the bus. Test results on this branch and on master are as reported in the description when using my test setup.